pwiz_tools/BiblioSpec/src/PepXMLreader.cpp

```diff
@@ -235,19 +235,20 @@ void PepXMLreader::startElement(const XML_Char* name, const XML_Char** attr)
 
        // handle msfragger source extensions for both native msfragger pepXMLs or PeptideProphet-analyzed pep.xmls
        if (search_engine_version.find("msfragger") == 0)
-       {
-           extensions.push_back("_uncalibrated.mgf");
-           extensions.push_back("_calibrated.mgf");
-
-           if (analysisType_ != MSFRAGGER_ANALYSIS)
+       {
+           // Prefer MGF over raw data for performance reasons
+           extensions.insert(extensions.begin(), "_calibrated.mgf");
+           extensions.insert(extensions.begin(), "_uncalibrated.mgf"); // Prefer uncalibrated, so place first in list
+           if (analysisType_ != MSFRAGGER_ANALYSIS)
                parentAnalysisType_ = MSFRAGGER_ANALYSIS;
        }
 
        setSpecFileName(fileroot_.c_str(), extensions, dirs);
 
-       if ((analysisType_ == MSFRAGGER_ANALYSIS || parentAnalysisType_ == MSFRAGGER_ANALYSIS) && bal::iends_with(getSpecFileName(), ".mgf")) {
-           lookUpBy_ = NAME_ID;
-           specReader_->setIdType(NAME_ID);
+       if ((analysisType_ == MSFRAGGER_ANALYSIS || parentAnalysisType_ == MSFRAGGER_ANALYSIS)) {
+           SPEC_ID_TYPE idType = bal::iends_with(getSpecFileName(), ".mgf") ? NAME_ID : INDEX_ID;

```

Comment:
This is probably breaking the non-TDF/non-MGF cases. For those cases, the correct idType gets set outside this block: so this block is only meant to override in the case of msfragger and MGF. I do think it's fine to insist on having the MGF if that was the input format though. Unlike the RAW/mzML case where they should be effectively interchangeable.

Reply:
So prefer MGF but not insist on it, and don't mess with the existing SPEC_ID_TYPE?

Reply:
>> I do think it's fine to insist on having the MGF if that was the input format though.
Not sure how to check that?

Reply:
The only way I can see to even guess that the pepXML came from TDF is if the spectrum_query elements have ion_mobility. But of course that might change if MSFragger adds support for FAIMS. Still, for now I would say if we have ion_mobility but the spectrum file we found doesn't end in `calibrated.mgf`, then we should abort.

Reply:
Also, how about a test? It might be reasonable to move the existing Hela test files for msfragger TDF into a Hela%20QC_PASEF_Slot1-5_01_57_cutout_2min.d directory in the inputs directory. Maybe add a dummy analysis.tdf file. Then it should break without these changes, right? (with a dummy TDF it would break differently than with a real one, but I don't think that's important)

Reply:
I don't know anything about MSFragger or any permutations of file formats it might employ - perhaps I should hand this off to you as the original implementer of this bit of code?

Reply:
Sounds good, I'll take over.